packages/flutter/lib/src/gestures/tap.dart

```diff
@@ -18,12 +18,15 @@ import 'recognizer.dart';
 class TapDownDetails {
   /// Creates details for a [GestureTapDownCallback].
   ///
-  /// The [globalPosition] argument must not be null.
-  TapDownDetails({ this.globalPosition = Offset.zero })
-    : assert(globalPosition != null);
+  /// Arguments [globalPosition] and [buttons] must not be null.
+  TapDownDetails({ this.globalPosition = Offset.zero, this.buttons = 0 })
+    : assert(globalPosition != null), assert(buttons != null);
 
   /// The global position at which the pointer contacted the screen.
   final Offset globalPosition;
+
+  /// The buttons pressed when the pointer contacted the screen.
+  final int buttons;
```

Comment:
I think for this to make sense we should require that it be a single button, not multiple buttons. Otherwise pressing left+right on a mouse will fail 99% of the time (because one will be detected a split second before the other) and work 1% of the time.

Reply:
I can limit it to "zero or one button" for https://github.com/flutter/flutter/pull/30339#discussion_r271087162. What do you think?

Reply:
I think it should be one button, and we should never have an event with zero buttons.